alerter.c

```diff
@@ -17,27 +17,33 @@ int returnCode = 200;
 }
 
 int networkAlertStub(float celcius) {
+	int returnCode = 200;
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
     // Return 200 for ok
     // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if(celcius > THRESHOLD_TEMP){
+    		returnCode = 500;
+    	}
+    else{
+    	returnCode = 200;
+    }
+    return returnCode;
 }
 
-void alertInCelcius(float farenheit) {
+int alertInCelcius(float farenheit) {
     float celcius = (farenheit - 32) * 5 / 9;
     int returnCode = networkAlert(celcius);
     if (returnCode != 200) {
         // let us keep a count of failures to report
-        alertFailureCount += 0;
+        alertFailureCount += 1;
     }
+    return alertFailureCount;
 }
 
 int main(void) {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
+    assert(alertFailureCount + 1 == alertInCelcius(400.5));

```

Comment:
These tests would miss the fact that `alertFailureCount` would need to 'count' the number of failures. So if we call `alertInCelcius(400.5)` twice, the count should be two.